v2/pkg/operator/local_stored_collector.go

```diff
@@ -220,17 +241,36 @@ func (o LocalStorageCollector) prepareM2DCopyBatch(log clog.PluggableLoggerInter
 		for _, img := range relatedImgs {
 			var src string
 			var dest string
-			imgSpec, err := image.ParseRef(img.Image)
-			if err != nil {
-				o.Log.Error("%s", err.Error())
-				return nil, err
-			}
-			src = imgSpec.ReferenceWithTransport
-
-			if imgSpec.IsImageByDigest() {
-				dest = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent + ":" + imgSpec.Digest[:hashTruncLen]}, "/")
+			if strings.HasPrefix(img.Image, ociProtocol) {
+				transportAndPath := strings.Split(img.Image, "://")
+				src = ociProtocolTrimmed + transportAndPath[1]
+				if len(transportAndPath) == 2 {
+					// generate a random hex string to use as a tag
+					var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
+					hex := strconv.FormatUint(rnd.Uint64(), 16)
+					var hexTag string
+					if len(hex) > hashTruncLen {
+						hexTag = hex[:hashTruncLen]
+					} else {
+						hexTag = hex
+					}
```

Comment:
Wouldn't this produce a different tag each time we run MirrorToDisk? 
If the user is not using targetTag, they might even have different tags in the enclave registry for the same image.
We could use something similar to [this](https://github.com/openshift/oc-mirror/blob/main/v2/pkg/archive/image-blob-gatherer.go#L38-L60) in order to get the digest of the image, and deduce the tag from there. WDYT?
Such a function might have its place in the `image` package?

Reply:
Yes, agree. Let's say the customer using this oci catalog wants to mirror only devworkspace-operator and one month later the customer decided to mirror also aws-load-balancer based on the same oci catalog on disk. This is going to generate two different tags even if the base oci catalog was the same one. 

I think it is safer to use the digest from the manifest of the oci catalog as suggested above and keep it in the image package.